lib/wallet/src/Cardano/Wallet/DB/Store/Delegations/Migration/Schema.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE EmptyDataDecls #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- |
+-- Copyright: © 2018-2023 IOHK
+-- License: Apache-2.0
+--
+-- Auto-generated Sqlite & Persistent machinery via Template-Haskell. This has
+-- been moved into a separate file so that we can treat it slightly differently
+-- when computing code-coverage.
+--
+-- More than 6K lines end-up being generated from the instructions below! As a
+-- result, we're going to ignore code-coverage on the following module and, no
+-- hand-written functions should be written in this module!
+
+module Cardano.Wallet.DB.Store.Delegations.Migration.Schema where
+
+import Prelude
+
+import Cardano.Pool.Types
+    ( PoolId )
+import Cardano.Slotting.Slot
+    ( SlotNo )
+import Cardano.Wallet.DB.Sqlite.Types
+    ( BlockId, sqlSettings' )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( UTCTime )
+import Database.Persist.TH
+    ( mkPersist, persistLowerCase, share )
+import GHC.Generics
+    ( Generic (..) )
+
+import qualified Cardano.Wallet.Primitive.Passphrase.Types as W
+import qualified Cardano.Wallet.Primitive.Types as W
+
+share
+    [ mkPersist sqlSettings'
+    ]
+    [persistLowerCase|
+
+
+-- Wallet IDs, address discovery state, and metadata.
+Wallet
+    walId                       W.WalletId                sql=wallet_id
+    walCreationTime             UTCTime                   sql=creation_time
+    walName                     Text                      sql=name
+    walPassphraseLastUpdatedAt  UTCTime Maybe             sql=passphrase_last_updated_at
+    walPassphraseScheme         W.PassphraseScheme Maybe  sql=passphrase_scheme
+    walGenesisHash              BlockId                   sql=genesis_hash
+    walGenesisStart             UTCTime                   sql=genesis_start
+
+    Primary walId
+    deriving Show Generic
+-- Track whether the wallet's stake key is registered or not.
+StakeKeyCertificate
+    stakeKeyCertWalletId             W.WalletId            sql=wallet_id
+    stakeKeyCertSlot                 SlotNo                sql=slot
+    stakeKeyCertType                 W.StakeKeyCertificate sql=type
+
+    Primary stakeKeyCertWalletId stakeKeyCertSlot
+    Foreign Wallet OnDeleteCascade stakeKeyRegistration stakeKeyCertWalletId
```

Comment:
```suggestion
```

The foreign key constraint requires us to define the `Wallet` table, which draws in more types that we would like to avoid drawing in. Since we only ever read from these tables — is it enough to lie a little and simply not mention the foreign key constraint here? The migration code should work regardless of whether the database has a foreign key constraint.

Reply:
I'm not sure I want to mess up with this. I think copy and paste is the right thing to do, whatever the price is.

Reply:
> I think copy and paste is the right thing to do, whatever the price is.

I tend to agree, but the trouble is the following: Imagine that in the future, we want to change or the `PassphraseScheme` type. Then we have to touch the schema defined here, but this schema is essentially dead code and it's about delegations, which is outside the scope of `PassphraseScheme`. At that point in time, we will be faced with the same question: Can we remove the `wallet_id` table from this schema if we also remove the foreign key constraints? The trouble is that at that time, we have much less context than we have now, so I would prefer to tackle this question now rather than later.